test/extended/builds/service.go

```diff
@@ -43,14 +42,20 @@ RUN curl -vvv hello-openshift:6379
 
 		g.Describe("with a build being created from new-build", func() {
 			g.It("should be able to run a build that references a cluster service", func() {
-				g.By("standing up a new hello world service")
-				err := oc.Run("new-app").Args("--name", "hello-openshift", k8simage.GetE2EImage(k8simage.Redis)).Execute()
+				g.By("standing up a new hello world nodejs service via oc new-app")
+				err := oc.Run("new-app").Args("nodejs~https://github.com/sclorg/nodejs-ex.git", "--name", "hello-nodejs").Execute()
```

Comment:
I've occasionally been curious on the side how the upcoming disconnected tests are going to deal with git cloning from the outside :-)

Reply:
Can this be "we build the image on the outside with the Git bits already installed, and then mirror the built image into the restricted-network environment before running the tests"?  That would make cloning-from-Git or other on-the-fly internet access technical debt that we'd have to clean out before we had the test-case green in restricted-networks where that internet access is not available.

Reply:
can - yes
do it in relatively minimal cycles like this change - no .... a much much much more involved change based on other recent build test image work we have had to wrt the disconnected e2e stuff

short term, e2e-aws-proxy needs to be unblocked

but as I mentioned to @adambkaplan in slack, our team needs to huddle up at some point and see how we will deal with the git cloning elements of our e2e's (which aside from testing git clone, test several of the upstream s2i base image implementations as part of vetting those)

Reply:
That's going to be a much bigger endeavor. "Clone source from git over the internet" is the most basic thing OCP builds do, and we do it everywhere in the build suite.

Is having a fully air-gapped CI test a goal? Or is the goal having CI tests run on a cluster that is restricted, but with some caveats. Example - proxy in a DMZ that has an allowlist of domains to access.